tools/undo-block/src/cli.rs

```diff
@@ -0,0 +1,75 @@
+use chrono::Utc;
+use near_chain::types::LatestKnown;
+use near_chain::{ChainStore, ChainStoreAccess, ChainStoreUpdate};
+use near_chain_configs::GenesisValidationMode;
+use near_primitives::block::Tip;
+use near_primitives::utils::to_timestamp;
+use near_store::{Mode, NodeStorage};
+use nearcore::load_config;
+use std::path::Path;
+
+#[derive(clap::Parser)]
+pub struct UndoBlockCommand {}
+
+impl UndoBlockCommand {
+    pub fn run(
+        self,
+        home_dir: &Path,
+        genesis_validation: GenesisValidationMode,
+    ) -> anyhow::Result<()> {
+        let near_config = load_config(home_dir, genesis_validation)
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+
+        let cold_config: Option<&near_store::StoreConfig> = near_config.config.cold_store.as_ref();
```

Comment:
I wonder if you need cold config at all. As you're only dealing with the head, it can't possibly affect cold storage.